### DISCUSSION TITLE
Custom completion with HTTP request example

### DISCUSSION THREAD

**Discussion by andrewbanchich** [nushell/nushell#6433] [Q&A]
I haven't been able to find any examples of this, but is there a way to cache the response of an expensive HTTP request used for custom completions?

My current script is making the request when I tab to auto complete, pulling down a bunch of results, and doing the same thing for every character typed. I just want it to make the request once it enters autocomplete mode, then search the results of that request on each keypress.

**Comment by herlon214** [comment]
I'm not sure if this is the best answer but definitely one option:

Currently we don't support caching for completion results, so you could use a file to create your own cache. At the first time you check if the file is created, if not you do the http req then save the result into a file. When the user is using the com completion again you read from the file instead of doing the request again.
That might also work if you store you value in the $env (but could affect nu's performance maybe).
